### PR TITLE
feat: get all collections for a given userId

### PR DESCRIPTION
**Pull request by Aaronel10**

### CODE REVIEW

File: web/src/pages/api/collection/all.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  body: {
+    userId: string;
+  };
+}
+
+export default async function Get(req: Request, res: NextApiResponse) {
+  const { userId } = req.body;
+
+  if (!userId) {
+    return res.status(402).json("No userId provided");
```
Comment (rob-3) [inline web/src/pages/api/collection/all.ts:15]: Is there a convention to send 402 when parameters aren't met? MDN says this means "payment required"

Comment (Rikimbili) [reply web/src/pages/api/collection/all.ts]: I'm pretty sure that if the body or some parameters are missing, we just send back 400 for the invalid request.

File: web/src/pages/api/collection/all.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import { NextApiRequest, NextApiResponse } from "next";
+
+import { prisma } from "../../../server/db/client";
+
+interface Request extends NextApiRequest {
+  body: {
+    userId: string;
```
Comment (Rikimbili) [inline web/src/pages/api/collection/all.ts:7]: We shouldn't be expecting to receive body for a GET endpoint (which I think it is). Instead, we should grab these from the query parameters.

I would recommend we add a field searchUserId or getUserId. Currently we are using userId to verify that you are the user you say you are when validating endpoints. If we want to grab some data for another userId then we have to add a new one.

File: web/src/pages/api/collection/all.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  };
+}
+
+export default async function Get(req: Request, res: NextApiResponse) {
+  const { userId } = req.body;
```
Comment (Rikimbili) [inline web/src/pages/api/collection/all.ts:12]: @rob-3 Shouldn't we be validating the request on all endpoints? If so we should add this here too

```
  // Validate if the user has a valid JWT token
  if (!(await validateRequest(req))) {
    return res.status(401).json({ error: "User not logged in." });
  }
```

Note that as soon as we do this, the userId will internally be verified to match the one present in the jwt. As such we need another parameter to specify the userId we want to fetch the collections for

**Review by Rikimbili — CHANGES_REQUESTED** [review]

File: web/src/pages/api/collection/all.ts (hunk 1 above)
Comment (Aaronel10) [reply web/src/pages/api/collection/all.ts]: switched error code to 400

**Review by Rikimbili — APPROVED** [review]

**Pull request merged**